build: make "gradlew eclipse" idempotent

### Description

Currently you can't run `gradlew eclipse` twice, on the second invocation:

```
lucene[main]$ ./gradlew eclipse
Starting a Gradle Daemon (subsequent builds will be faster)
> Task :eclipseClasspath FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':eclipseClasspath'.
> Cannot parse file '/home/rmuir/workspace/lucene/.classpath'.
         Perhaps this file was tinkered with? In that case try delete this file and then retry.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 22s
5 actionable tasks: 5 executed
```

It only seems to "protect" `.classpath` and not `.settings` or the other output files, I think?

It is a minor annoyance to a human, but unfriendly to automation.

As a workaround you can do `rm -f .classpath && ./gradlew eclipse` for now. Have not dug into it, but the word `tinkered` does not exist in our codebase, so the problem is coming from somewhere else :)

--- a/build-tools/build-infra/src/main/java/org/apache/lucene/gradle/plugins/ide/EclipseSupportPlugin.java
+++ b/build-tools/build-infra/src/main/java/org/apache/lucene/gradle/plugins/ide/EclipseSupportPlugin.java
@@ -37,6 +37,7 @@ import org.gradle.api.artifacts.VersionCatalog;
 import org.gradle.api.plugins.JavaPlugin;
 import org.gradle.api.plugins.JavaPluginExtension;
 import org.gradle.api.provider.Provider;
+import org.gradle.api.tasks.Delete;
 import org.gradle.api.tasks.Sync;
 import org.gradle.plugins.ide.eclipse.EclipsePlugin;
 import org.gradle.plugins.ide.eclipse.GenerateEclipseClasspath;
@@ -153,6 +154,15 @@ public class EclipseSupportPlugin extends LuceneGradlePlugin {
               task.dependsOn(luceneEclipseJdt);
             });
 
+    var wipePreviousConfiguration =
+        tasks.register(
+            "wipePreviousConfiguration",
+            Delete.class,
+            task -> {
+              Path rootDir = getProjectRootPath(project);
+              task.delete(rootDir.resolve(".classpath"), rootDir.resolve(".project"));
+            });
+
     // Add gradle plugin portal to the source repository list and
     // apply any ecj source repository hackery the same way as everywhere.
     project.getRepositories().gradlePluginPortal();
@@ -163,6 +173,8 @@ public class EclipseSupportPlugin extends LuceneGradlePlugin {
         .named("eclipseClasspath")
         .configure(
             task -> {
+              task.dependsOn(wipePreviousConfiguration);
+
               var classpath = task.getClasspath();
               classpath.setDefaultOutputDir(project.file("build/eclipse"));
               classpath.setDownloadJavadoc(false);

